core/src/main/java/io/cloudevents/core/v1/CloudEventBuilder.java

```diff
@@ -119,7 +121,12 @@ public CloudEvent build() {
             throw createMissingAttributeException(TYPE);
         }
 
-        return new CloudEventV1(id, source, type, datacontenttype, dataschema, subject, time, this.data, this.extensions);
+        CloudEvent cloudEvent = new CloudEventV1(id, source, type, datacontenttype, dataschema, subject, time, this.data, this.extensions);
+
+        CloudEventValidatorProvider validator = CloudEventValidatorProvider.getInstance();
+        validator.validate(cloudEvent);
```

Comment:
We need to add the same logic for the 0.3 builder https://github.com/cloudevents/sdk-java/blob/main/core/src/main/java/io/cloudevents/core/v03/CloudEventBuilder.java